packages/block-library/src/separator/style.scss

```diff
@@ -8,7 +8,9 @@
 
 	// Dots style
 	&.is-style-dots {
-		background: none; // Override any background themes often set on the hr tag for this style.
+		// Override any background themes often set on the hr tag for this style.
+		// also override the color set in the editor since it's intented for normal HR
+		background: none !important;
```

Comment:
I see there was some discussion about the use of `!important` here in the pull request comments, and separately a fix addressing a TwentyTwenty-specific incompatibility at [Trac#47811](https://core.trac.wordpress.org/ticket/47811).

Which leaves the question: Do we actually need it? Can it be removed?

See also: #19539

Alternatively, I wonder:

- Only including this specific style rule if `.has-text-color` and/or `.has-background`, representative of the fact that the default should only need to be reset when the user chooses to apply a custom color (correct me if I'm mistaken)
- Can it be addressed by assuring that the [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) of this selector would always "defeat" a theme's styling? Especially considering we would probably want this reset only in the case of applying when a custom color is used, i.e. adding `.has-text-color` and/or `.has-background` as additional modifier classes:

```
.wp-block-separator.is-style-dots.has-background,
.wp-block-separator.is-style-dots.has-text-color {
	background: none;
}
```

Reply:
What if a different method for generating dots was used?
The current method is a lot of CSS, that is a hassle for themes to override. I was going for the minimal style of making dots with multiple background images using radial gradients, but the current style using !important wouldn't let me.
I also couldn't use a dotted border because all those other styles I would have to override.

So what if the core style is the radial gradient instead of an important no background?

Reply:
It'd be worth exploring a simpler approach.